sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -353,11 +353,13 @@ class SpillableHostBufferHandle private (
       if (closed) {
         throw new IllegalStateException(
           "attempting to materialize a closed handle")
+      // after spilling, the host can get removed asynchronously, so let's
```

Comment:
Can we have a follow on issue to cache the materialized version after reading it in again? The issue is that if we spill to disk once, then at that point each time we want to read that data again, even if a copy is sitting in host memory, we have to read it from disk again and add another copy of it to host memory.

Reply:
Is this the equivalent of the concept of "unspilling"?

Reply:
yes, but not clear if we are going to keep the disk copy around to re-spill later or not. We should discuss that

Reply:
filed https://github.com/NVIDIA/spark-rapids/issues/12026 as a placeholder